lib/passes/analysis/MemOpVisitor.cpp

```diff
@@ -61,11 +71,22 @@ void MemOpVisitor::visitCallBase(llvm::CallBase& cb) {
   }
 }
 
-void MemOpVisitor::visitMallocLike(llvm::CallBase& ci, MemOpKind k) {
-  //  LOG_DEBUG("Found malloc-like: " << ci.getCalledFunction()->getName());
+template <class T>
+auto expectSingleUser(llvm::Value* value) -> T* {
+  auto users    = value->users();
+  auto users_it = users.begin();
+  auto first    = *users_it++;
+  assert(users_it == users.end());
+  assert(isa<T>(first));
+  return dyn_cast<T>(first);
+}
 
-  SmallPtrSet<BitCastInst*, 4> bcasts;
+using MallocGeps   = SmallPtrSet<GetElementPtrInst*, 2>;
+using MallocBcasts = SmallPtrSet<BitCastInst*, 4>;
 
+auto collectRelevantMallocUsers(llvm::CallBase& ci) -> std::pair<MallocGeps, MallocBcasts> {
```

Comment:
Trailing return types (`auto foo() -> bar{..}`) is not common in the code base.
I suggest to keep it as `bar foo() {..}`.